src/task_types/resmoke_tasks.rs

```diff
@@ -320,13 +339,61 @@ impl GenResmokeTaskServiceImpl {
 
         if !current_tests.is_empty() {
             sub_suites.push(SubSuite {
-                name: name_generated_task(&params.task_name, Some(i), Some(n_suites as u64)),
+                index: Some(i),
+                name: params.task_name.to_string(),
                 test_list: current_tests,
             });
         }
 
         Ok(sub_suites)
     }
+
+    /// Create version of the generated sub-tasks for all the multiversion combinations.
+    ///
+    /// # Arguments
+    ///
+    /// * `params` - Parameters for how task should be generated.
+    /// * `sub_suites` - Sub-suites that were created for the task being generated.
+    ///
+    /// # Returns
+    ///
+    /// List of sub-suites that includes versions fall all multiversion combinations.
+    fn create_multiversion_combinations(
+        &self,
+        params: &ResmokeGenParams,
+        sub_suites: &[SubSuite],
+    ) -> Result<Vec<SubSuite>> {
```

Comment:
Here is where we are creating the configuration to actually generate the multiversion tasks. We will take the split up sub-tasks and create multiversion sub-tasks for each of them based on the multiversion configuration.